guardian/grid
# Add new programme usage rights

## What does this change?

This PR (co-authored by @Conalb97) introduces 3 new programme usage rights meant to replace the "Programme Promotional" usage right that was introduced by this PR #3774.

These new programme rights are:
* `Programmes - Organisation Owned` - for all images that are related to a programme and have the organisation as the credit.
* `Programmes - Independents` -  for all images that are related to a programme and are either made by an independent production company, the organisation or a joint production.
* `Programmes - Acquisitions` - for all images of programmes that are bought (acquired) by the organisation for a period of time.

Adding the new usage rights to the `usageRights.applicable` configuration should allow a user to select either in usage rights edit form and allow image(s) to be searched using the `+category: programme-organisation-owned` or `+category: programme-independents` or `+category: programme-acquisitions` chip.

### Applying new programme usage rights to an image

![Select and save programme usage rights](https://github.com/guardian/grid/assets/12212239/889b43f0-0c2e-4d49-838d-d34f04ee0fff)

### Searching using new programme usage rights / categories

![Search using programme usage rights](https://github.com/guardian/grid/assets/12212239/ec1ede38-6c44-4575-8a41-ace738fc0568)

The objective of this PR and a follow-up PR, is to replace / deprecate the "Programme Promotional" usage rights. Before the "Programme Promotional" usages rights is deprecated, images with the "Programme Promotional" usage rights should be reassigned the new programme usage rights where applicable.

To deploy these new image rights, an update to the Elasticsearch mapping and a migration (if there are existing documents in Elasticsearch) has to be done to index the "Independent Type" and "Production Company" fields in the "[ProgrammeIndependents](https://github.com/bbc/grid/blob/82481202f6d73273c57ee0eeebf7187c748a23dc/common-lib/src/main/scala/com/gu/mediaservice/model/UsageRights.scala#L582)" usage rights model.

## How should a reviewer test this change?

Add the new usage rights to the `usageRights.applicable` list and this should add the new usage rights to the "Usage rights" select box and add them to the `category:` search chip.

```hocon
usageRights.applicable = [
  "com.gu.mediaservice.model.ProgrammesOrganisationOwned",
  "com.gu.mediaservice.model.ProgrammesIndependents",
  "com.gu.mediaservice.model.ProgrammesAcquisitions"
]
```

Add these new configurations `programmesOrganisationOwned`, `programmesIndependents` and `programmesAcquisitions` in the `application.conf` file to configure custom descriptions of the usage rights and options to select from that should appear on the "usage rights edit form".

```hocon
usageRightsConfigProvider = {
  className: "com.gu.mediaservice.lib.config.RuntimeUsageRightsConfig"
  config {
    programmesOrganisationOwned {
      description = "This is a configurable description of the usage right." # Optional config - if not sensible default description is used
    }
    programmesIndependents {
      description = "This is a configurable description of the usage right." # Optional config - if not sensible default description is used
      independentTypes = [
        {
          name = "BBC Studios"
          productionsCompanies = ["Studio 1", "Studio 2"]
        }
        {
          name = "Independent"
          productionsCompanies = ["A", "B", "C"]
        }
        {
          name = "Joint Production"
          productionsCompanies = ["Joint Prod 1", "Joint Prod 2"]
        }
      ]
    }
    programmesAcquisitions {
      description = "This is a configurable description of the usage right." # Optional config - if not sensible default description is used
    }
  }
}
```

## How can success be measured?

* `Programme Organisation Owned`, `Programme Independents` and `Programme Acquisitions` should be selectable on the usage rights edit form.
* Custom usage right descriptions should be rendered on the usage rights edit form if defined. If not sensible defaults  should rendered instead.
* The new usage rights should be persisted (in DynamoDB and Elasticsearch) when applied to an image.
* `programme-organisation-owned`, `programme-independents` and `programme-acquisitions` should appear as options for the `category` chip.

## Who should look at this?

@guardian/digital-cms

## Tested? Documented?
